### PR TITLE
provider/aws: Add multi-region option to cloudtrail

### PR DESCRIPTION
**Pull request by Bowbaq**
Adds the option to enable a multi-region trail.

### CODE REVIEW

File: builtin/providers/aws/resource_aws_cloudtrail.go [hunk 1]
```diff
@@ -73,6 +78,9 @@ func resourceAwsCloudTrailCreate(d *schema.ResourceData, meta interface{}) error
 	if v, ok := d.GetOk("include_global_service_events"); ok {
 		input.IncludeGlobalServiceEvents = aws.Bool(v.(bool))
 	}
+	if v, ok := d.GetOk("is_multi_region_trail"); ok {
+		input.IncludeGlobalServiceEvents = aws.Bool(v.(bool))
```
Comment (rowleyaj) [inline builtin/providers/aws/resource_aws_cloudtrail.go]: Is this a copy paste error?

Comment (Bowbaq) [inline builtin/providers/aws/resource_aws_cloudtrail.go]: indeed, good catch

File: builtin/providers/aws/resource_aws_cloudtrail_test.go [hunk 2]
```diff
@@ -193,6 +226,7 @@ resource "aws_cloudtrail" "foobar" {
     s3_bucket_name = "${aws_s3_bucket.foo.id}"
     s3_key_prefix = "/prefix"
     include_global_service_events = false
+    is_multi_region_trail = true
```
Comment (radeksimko) [inline builtin/providers/aws/resource_aws_cloudtrail_test.go]: Can you please omit this change in the data for the existing tests? It's breaking the existing two test cases:

```
=== RUN   TestAccAWSCloudTrail_basic
--- FAIL: TestAccAWSCloudTrail_basic (26.90s)
	testing.go:148: Step 1 error: Error applying: 1 error(s) occurred:

		* aws_cloudtrail.foobar: InvalidParameterCombinationException: Multi-Region trail must include global service events.
			status code: 400, request id: *REDACTED*
=== RUN   TestAccAWSCloudTrail_enable_logging
--- FAIL: TestAccAWSCloudTrail_enable_logging (27.56s)
	testing.go:148: Step 1 error: Error applying: 1 error(s) occurred:

		* aws_cloudtrail.foobar: InvalidParameterCombinationException: Multi-Region trail must include global service events.
			status code: 400, request id: *REDACTED*
```

File: builtin/providers/aws/resource_aws_cloudtrail_test.go [hunk 3]
```diff
@@ -227,3 +261,42 @@ resource "aws_s3_bucket" "foo" {
...
+				"StringEquals": {
+					"s3:x-amz-acl": "bucket-owner-full-control"
+				}
+			}
+		}
+	]
+}
+POLICY
+}
+`, cloudTrailRandInt, cloudTrailRandInt, cloudTrailRandInt)
```
Comment (radeksimko) [inline builtin/providers/aws/resource_aws_cloudtrail_test.go]: :+1: Thanks for adding this test case.

File: website/source/docs/providers/aws/r/cloudtrail.html.markdown [hunk 4]
```diff
@@ -71,6 +71,8 @@ The following arguments are supported:
     Setting this to `false` will pause logging.
 * `include_global_service_events` - (Optional) Specifies whether the trail is publishing events
     from global services such as IAM to the log files. Defaults to `true`.
+* `is_multi_region_trail` - (Optional) // Specifies whether the trail is created in the current
```
Comment (radeksimko) [inline website/source/docs/providers/aws/r/cloudtrail.html.markdown]: Is there any reason for those slashes here? `//` :wink: If not, please remove those.

**Comment by radeksimko** [comment]
There is a few (IMO small) things to fix which I mentioned above, otherwise it's looking good @Bowbaq

**Comment by Bowbaq** [comment]
@radeksimko will fix, is there a way to make the acceptance tests use the credentials in `~/.aws/credentials` instead of environment variables?

**Comment by radeksimko** [comment]
> is there a way to make the acceptance tests use the credentials in ~/.aws/credentials instead of environment variables?

I'm afraid there isn't atm. I was also thinking of switching from ENV variables to a specific profile name, now when it's supported. I created a separate issue to discuss this: https://github.com/hashicorp/terraform/issues/5019

**Pull request merged**